### PR TITLE
Fix restricted transfer check

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: test/token/restricted.sol [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        token.file("memberlist", address(memberlist));
+    }
+
+    function testTransferFromTokensToMemberWorks(uint256 amount, address targetUser, uint256 validUntil) public {
+        vm.assume(validUntil > block.timestamp && targetUser != address(0) && targetUser != address(this));
+
+        memberlist.updateMember(targetUser, validUntil);
+        assertEq(memberlist.members(targetUser), validUntil);
+
+        token.mint(address(this), amount);
```
Comment (AStox) [inline test/token/restricted.sol:34]: mint is an `auth` function but still allows you to mint to a non-whitelisted address. Is this okay?

**Review by AStox — DISMISSED** [review]
Left one question, but otherwise looks good

File: test/token/restricted.sol [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        assertEq(memberlist.members(targetUser), block.timestamp);
+
+        vm.warp(block.timestamp + 1);
+
+        token.mint(address(this), amount);
+        vm.expectRevert(bytes("RestrictedToken/not-allowed-to-hold-token"));
+        token.transfer(targetUser, amount);
+        assertEq(token.balanceOf(targetUser), 0);
+    }
+}
```
Comment (NunoAlexandre) [inline test/token/restricted.sol:95]: Very clean tests, thanks 🌟

**Review by NunoAlexandre — APPROVED** [review]
Looks good 👌

File: test/token/restricted.sol (hunk 1 above)
Comment (offerijns) [reply test/token/restricted.sol:34]: Also discussed this a bit with Manuel. The only possible reason could be some kind of emergency, but even then if the origin is a ward that can mint, they can also add a member. So I agree and will add it.

**Pull request merged**